src/plugins/BrowserDetect.ts

```diff
@@ -0,0 +1,60 @@
+export const browserDetect = () => {
+    const browserAgent = navigator.userAgent
+    let browserName = navigator.appName
+    let browserVersion = '' + parseFloat(navigator.appVersion)
+    let browserMajorVersion: number
+    let Offset, OffsetVersion, ix
+
+    // For Chrome
+    if ((OffsetVersion = browserAgent.indexOf('Chrome')) != -1) {
+        browserName = 'Chrome'
+        browserVersion = browserAgent.substring(OffsetVersion + 7)
+    }
+
+    // For Microsoft internet explorer
+    else if ((OffsetVersion = browserAgent.indexOf('MSIE')) != -1) {
+        browserName = 'Microsoft Internet Explorer'
+        browserVersion = browserAgent.substring(OffsetVersion + 5)
+    }
+
+    // For Firefox
+    else if ((OffsetVersion = browserAgent.indexOf('Firefox')) != -1) {
+        browserName = 'Firefox'
```

Comment:
no browserVersion is read for firefox. Is this intended?

Reply:
in firefox, this line should be enough:
```
let browserVersion = '' + parseFloat(navigator.appVersion)
```